contracts/DPublish.sol

```diff
@@ -1,15 +1,43 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.2;
 
+struct Manuscript {
+    address[] authors;
+}
+
 contract DPublish {
-    mapping(string => address) public submitted_manuscripts;
+    mapping(string => Manuscript) submitted_manuscripts;
 
     /**
-    * Submit manuscript
+     * Submit manuscript
      */
     function submit_manuscript(string calldata idmanuscript) public {
-        submitted_manuscripts[idmanuscript] = msg.sender;
+        require(
+            submitted_manuscripts[idmanuscript].authors.length == 0,
+            "Manuscript already exists"
+        );
+
+        submitted_manuscripts[idmanuscript].authors.push(msg.sender);
     }
 
+    function add_author(string calldata idmanuscript, address author) public {
+        require(
+            submitted_manuscripts[idmanuscript].authors.length > 0,
+            "Manuscript does not exist"
+        );
+
+        require(
+            submitted_manuscripts[idmanuscript].authors[0] == msg.sender,
+            "Only the first author can add authors"
+        );
 
-}
\ No newline at end of file
+        for (uint256 i = 0; i < submitted_manuscripts[idmanuscript].authors.length; i++) {
```

Comment:
Loops are very expensive in terms of gas.
in the manuscript struct you can define authors as a mapping
```solidity
mapping (address => bool) public authors;
```
Then you would simply read the author's address from the mapping. if it returns `true` it's because it exists. This works because solidity mappings initialize all possible keys as `0/false`.